Bio/Seq.py

```diff
@@ -1503,30 +1561,26 @@ def translate(
     ):
         """Translate an unknown nucleotide sequence into an unknown protein.
 
-        e.g.
+        If your sequence makes sense as codons (e.g. a poly-A tail AAAAAA),
+        it will be translated accordingly:
 
-        >>> my_seq = UnknownSeq(9, character="N")
-        >>> print(my_seq)
-        NNNNNNNNN
-        >>> my_protein = my_seq.translate()
-        >>> my_protein
-        UnknownSeq(3, character='X')
-        >>> print(my_protein)
-        XXX
+        >>> UnknownSeq(7, character='A').translate()
+        UnknownSeq(2, character='K')
 
-        In comparison, using a normal Seq object:
-
-        >>> my_seq = Seq("NNNNNNNNN")
-        >>> print(my_seq)
-        NNNNNNNNN
-        >>> my_protein = my_seq.translate()
-        >>> my_protein
-        Seq('XXX')
-        >>> print(my_protein)
-        XXX
+        Otherwise, it will be translated as X for unknown amino acid:
 
+        >>> UnknownSeq(7).translate()
+        UnknownSeq(2, character='X')
         """
-        return UnknownSeq(self._length // 3, character="X")
+        try:
+            s = Seq(self._character * 3).translate(
+                table=table, stop_symbol=stop_symbol, to_stop=to_stop, cds=cds, gap=gap
+            )
+        except CodonTable.TranslationError:
+            # Preserve historic behaviour, ??? (default character) and XXX -> X
+            s = "X"
+        # Don't worry about to_stop - no known stop codon is three bases the same,
```

Comment:
I believe this is for `Bio.codonalign` only?

Reply:
I've referring to the standard ``.translate(...)`` method and the free standing function, which have a ``to_stop=True`` option to stop translating at the first in frame stop codon.

Reply:
Sorry, I meant the `CodonTable.TranslationError`

Reply:
No, that's the exception the translate method may throw, defined here:
https://github.com/biopython/biopython/blob/biopython-178/Bio/Data/CodonTable.py#L43

Reply:
OK